jug.md

```diff
@@ -98,18 +98,22 @@ Jug Semantics
 ```k
     syntax JugAuthStep ::= "init" String
  // ------------------------------------
-    rule <k> Jug . init ILK => . ... </k>
+    rule <k> Jug . init ILK_ID => . ... </k>
          <current-time> NOW </current-time>
-         <jug-ilks> ... ILK |-> Ilk ( ... duty: ray(0) => ray(1), rho: _ => NOW ) ... </jug-ilks>
+         <jug-ilks> ... ILK_ID |-> Ilk ( ... duty: ray(0) => ray(1), rho: _ => NOW ) ... </jug-ilks>
+
+    rule <k> Jug . init ILK_ID ... </k>
+         <jug-ilks> JUG_ILKS => JUG_ILKS [ ILK_ID <- Ilk ( ... duty: ray(0) , rho: 0 ) ] </jug-ilks>
```

Comment:
I'm a little confused by the syntax here (not saying it's wrong, just wondering why things are the way they are). Naively, I'd expect something like:
```
<jug-ilks> JUG_ILKS => JUG_ILKS ( ILK_ID |-> Ilk (... duty: ray(0) , rho: 0 ) ) </jug-ilks>
```
How is the `[ K <- V ]` syntax different from the `K |-> V` syntax?

Reply:
The operator `_Map_` is map union, and `K |-> V` is a constructor of sort `Map`, so when you do `MAP => MAP (K |-> V)` (which looks like `MAP => _Map_(MAP, _|->_(K, V))` in AST form), you're asserting that the maps `MAP` and `(K |-> V)` have disjoint keys. _If_ that assumption is broken, then it's undefined (entire thing is #False in Haskell backend, and runtime error in LLVM backend).

The operator `_[_<-_]` is map update, so if you do `MAP => MAP [ K <- V ]` and `MAP` already contains key `K`, it will just override that entry in `MAP` with `K |-> V`, instead of crashing.